### PR TITLE
workflows:pin node versions to lts

### PR DESCRIPTION
**Pull request by gurgunday**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.1.0
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.2.1
```
Comment (gurgunday) [inline .github/workflows/ci.yml:20]:
```suggestion
    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v5.0.0
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.1.0
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v5.0.0
     with:
       license-check: true
+      node-versions: '["16", "18", "20", "22"]'
```
Comment (gurgunday) [inline .github/workflows/ci.yml:23]: Should we test on node 16?

Comment (mcollina) [reply .github/workflows/ci.yml:23]: also on v14, I don't think we have shipped a major or this yet.

Comment (gurgunday) [inline .github/workflows/ci.yml:23]:
```suggestion
      node-versions: '["14", "16", "18", "20", "22"]'
```

Comment (gurgunday) [reply .github/workflows/ci.yml]: We might need tape, will take a look

**Review by Fdawgs — APPROVED** [review]

**Comment by mcollina** [comment]
Seems like I spoke too early. Let’s cut it at v20+ and cut a major then.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,6 +17,7 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.1.0
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v5.0.0
     with:
       license-check: true
+      node-versions: '["14", "16", "18", "20", "22"]'
```
Comment (gurgunday) [inline .github/workflows/ci.yml:23]:
```suggestion
      node-versions: '["20", "22"]'
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**